aiida_quantumespresso/utils/bands.py

```diff
@@ -3,11 +3,25 @@
 from __future__ import absolute_import
 
 
-def get_highest_occupied_band(bands, threshold=0.01):
+def get_highest_occupied_band(bands, threshold=0.005):
     """Retun the index of the highest-occupied molecular orbital.
 
     The expected structure of the bands node is the following:
 
+        * an array called `occupations`
+        * with 3 dimensions if spin polarized, otherwise 2 dimensions
+        * dimensions loop over, spin channel, kpoints, bands
+
+    .. note::
+
+        The threshold is used both as a limit below which a band is considered as unoccupied as well as a measure of
+        numerical noise in the occupancies. The first band to have an occupancy below the threshold at all kpoints is
+        marked as the LUMO. All subsequent bands are then expected to have an occupation that is less than twice the
+        threshold. The reason for the factor two is that in the worst case when the LUMO has an occupation exactly equal
+        to the threshold and a subsequent has twice that value, we can still consider that as not to break the
+        monotonically decreasing behavior of occupations. It can be considered as having the exact same value (equal to
```

Comment:
I think we shouldn't talk about "monotonicity" here - all that's checked is that the bands above LUMO are _also_ unoccupied (with a less strict threshold).

The 
``:raises ValueError: if the occupations are not monotonically decreasing at all k-points between LUMO and HOMO`` 
could probably be changed to something like 
``:raises ValueError: if any occupation above LUMO exceeds `2 * threshold``

Reply:
Good point. Fixed the docstrings and commit message, and added a unit test for the threshold parameter as well.